tests/support/__init__.py

```diff
@@ -296,12 +306,24 @@ def fixturefile(relative_path, fixture_format=None):
     #_, extension = os.path.splitext(os.path.basename(tmp_path))
     #assert fixture_format == extension, "fixture file does not match format"
 
-    if fixture_format == 'json':
-        return _fixturefile_json(tmp_path)
+    data = None
+
+    if fixture_format == 'binary':
+        with open(tmp_path, 'rb') as binfile:
+            data = binfile.read()
+    elif fixture_format == 'json':
+        data = _fixturefile_json(tmp_path)
     else:
         raise AssertionError(
             "unsupported fixture format: %s" % (fixture_format,))
 
+    return (data, tmp_path) if include_path else data
+
+
+def fixturefile_normname(relative_path, prefix=None):
+    normname, _ = relative_path.split('--')
```

Comment:
From experience I know that splitting strings and silently assuming a fixed number of resulting parts like this is a bit fragile and tends to break sooner or later. I'd suggest adding a `maxsplit=1` arg to split or use `normname=blabla[0] if additional parts are _always_ irrelevant. Otherwise explicitly assert exactly one occurrence of `'--'` or split and check part count before assignment.